tests/src/junit/java/net/neoforged/neoforge/unittest/transfer/InstancedHandlerTests.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @Test
+    public void endlessHandlers() {
+        //InfiniteResourceHandlers creates infinite of a specified resource, but doesn't allow insertion
+        testEndlessResource(FluidResource.of(Fluids.WATER), FluidResource.EMPTY);
+        testEndlessResource(ItemResource.of(Blocks.COBBLESTONE), ItemResource.EMPTY);
+    }
+
+    private static <T extends IResource> void testVoidResource(VoidResourceHandler<T> handler, T emptyResource) {
+        Assertions.assertThat(handler.size()).withFailMessage("Size should be 1").isEqualTo(1);
+        Assertions.assertThat(handler.supportsExtraction()).withFailMessage("Extraction should be not be allowed").isFalse().isEqualTo(handler.supportsExtraction(0));
+        Assertions.assertThat(handler.supportsInsertion()).withFailMessage("Insertion should be be allowed").isTrue().isEqualTo(handler.supportsInsertion(0));
+        Assertions.assertThat(ResourceHandlerUtil.isValid(handler, emptyResource)).withFailMessage("Voids should always be valid for a resource").isTrue();
+        Assertions.assertThat(handler.getCapacity(0, emptyResource)).withFailMessage("Voids should hava capacity of max int").isEqualTo(Integer.MAX_VALUE);
+        Assertions.assertThat(handler.getResource(0)).withFailMessage("Empty resource should match").isEqualTo(emptyResource);
+
+        try (var transaction = TransactionManager.open(null)) {
+            Assertions.assertThat(Integer.MAX_VALUE)
+                    .withFailMessage("Void should be able to accept infinite resources")
+                    .isEqualTo(handler.insert(0, emptyResource, Integer.MAX_VALUE, transaction))
+                    .isEqualTo(handler.insert(emptyResource, Integer.MAX_VALUE, transaction));
+
+            Assertions.assertThat(0)
+                    .withFailMessage("Void should be provide no resources")
+                    .isEqualTo(handler.extract(0, emptyResource, Integer.MAX_VALUE, transaction))
+                    .isEqualTo(handler.extract(emptyResource, Integer.MAX_VALUE, transaction));
+        }
+    }
+
+    private static <T extends IResource> void testEndlessResource(T resource, T emptyResource) {
+        InfiniteResourceHandler<T> handler = new InfiniteResourceHandler<>(resource);
+        Assertions.assertThat(handler.size()).withFailMessage("Size should be 1").isEqualTo(1);
+        Assertions.assertThat(handler.supportsExtraction()).withFailMessage("Extraction should be allowed").isTrue().isEqualTo(handler.supportsExtraction(0));
+        Assertions.assertThat(handler.supportsInsertion()).withFailMessage("Insertion should not be allowed").isFalse().isEqualTo(handler.supportsInsertion(0));
+        Assertions.assertThat(ResourceHandlerUtil.isValid(handler, resource)).withFailMessage("Endless resource can not be inserted into, thus isValid should be false").isFalse();
+        Assertions.assertThat(Integer.MAX_VALUE).withFailMessage("Capacity should be max int")
```

Comment:
We should also check the stored amount, and the long variations of these methods (being max long)